src/wallet/api.rs

```diff
@@ -1502,78 +1475,43 @@ impl Wallet {
         pubkey2: &PublicKey,
-        address_label: &String,
+        _address_label: &str,
     ) -> Result<(), WalletError> {
         let descriptor = self
             .rpc
-            .get_descriptor_info(&format!("wsh(sortedmulti(2,{},{}))", pubkey1, pubkey2))?
-            .descriptor;
-        let result = self
+            .get_descriptor_info(&format!("wsh(sortedmulti(2,{},{}))", pubkey1, pubkey2))?;
+        let import_requests = Value::Array(vec![
+            json!({"desc":descriptor.descriptor,"timestamp":"now"}),
+        ]);
+        let _res: Vec<Value> = self
             .rpc
-            .import_multi(
-                &[ImportMultiRequest {
-                    timestamp: Timestamp::Now,
-                    descriptor: Some(&descriptor),
-                    watchonly: Some(true),
-                    label: Some(address_label),
-                    ..Default::default()
-                }],
-                Some(
-                    &(ImportMultiOptions {
-                        rescan: Some(false),
-                    }),
-                ),
-            )
+            .call("importdescriptors", &[import_requests])
             .unwrap();
-        for r in result {
-            if !r.success {
-                log::warn!(target: "Wallet:import_addresses", "{:?}", r.warnings);
-                if let Some(e) = r.error {
-                    return Err(WalletError::Protocol(e.message));
-                }
-            }
-        }
         Ok(())
     }
 
     /// Imports a redeem script into the wallet.
     pub fn import_redeemscript(
         &self,
         redeemscript: &ScriptBuf,
-        address_label: &String,
+        _address_label: &str,
     ) -> Result<(), WalletError> {
         let spk = redeemscript_to_scriptpubkey(redeemscript);
-        let result = self.rpc.import_multi(
-            &[ImportMultiRequest {
-                timestamp: Timestamp::Now,
-                script_pubkey: Some(ImportMultiRequestScriptPubkey::Script(&spk)),
-                redeem_script: Some(redeemscript),
-                watchonly: Some(true),
-                label: Some(address_label),
-                ..Default::default()
-            }],
-            Some(
-                &(ImportMultiOptions {
-                    rescan: Some(false),
-                }),
-            ),
-        )?;
-        for r in result {
-            if !r.success {
-                log::warn!(target: "Wallet:import_addresses", "{:?}", r.warnings);
-                if let Some(e) = r.error {
-                    return Err(WalletError::Protocol(e.message));
-                }
-            }
-        }
+        let descriptor = self
+            .rpc
+            .get_descriptor_info(&format!("raw({:x})", spk))
+            .unwrap()
+            .descriptor;
+        let import_requests = Value::Array(vec![json!({"desc":descriptor,"timestamp":"now"})]);
+        let _res: Vec<Value> = self.rpc.call("importdescriptors", &[import_requests])?;
```

Comment:
I don't see why we would need multiple types of import calls. Thats more risky and unclean code. These are legacy parts of the wallet so are remnants of the old architecture.

Let's combine all import calls into a single wallet-level automated `Wallet::import_descriptors(&self)` API. 

The wallet already knows at each point in time what all the descriptors to import.

Reply:
closing this then, and taking this up in #135